src/learner.cc

```diff
@@ -713,6 +689,7 @@ class LearnerImpl : public Learner {
       CHECK_LE(num_col, static_cast<uint64_t>(std::numeric_limits<unsigned>::max()))
         << "Unfortunately, XGBoost does not support data matrices with "
         << std::numeric_limits<unsigned>::max() << " features or greater";
+      // FIXME(trivialfis): Do we need this max since num_feature is just unsigned 0?
```

Comment:
Here, we are computing the maximum number of features over multiple cached DMatrix objects.